### ISSUE TITLE
Log an explicit error message when the server returns a 414 URI Too Long status

### ISSUE THREAD

**Issue by Mogztter** [Mogztter/asciidoctor-kroki#224]
Most Web servers have a URI limit to prevent Denial-of-Service or malicious attacks (i.e., buffer overflow/high memory usage...).

According to [RFC 2616 (Hypertext Transfer Protocol HTTP/1.1) section 3.2.1](http://www.faqs.org/rfcs/rfc2616.html), the server should return [`414 (Request-URI Too Long)`](https://developer.mozilla.org/fr/docs/Web/HTTP/Status/414) status if a URI is longer than the server can handle.

It's also important to note that browsers also have [a URI limit](https://stackoverflow.com/questions/417142/what-is-the-maximum-length-of-a-url-in-different-browsers/417184#417184). Having said that, and if we exclude IE, the lowest value is 10240 on Edge 16. This value is relatively high and we should not exceed it.
For reference, the current URI limit on https://kroki.io is 4000.

Anyway, if the Kroki server returns a 414 URI Too Long status, we could log an explicit error message to suggest how to resolve this issue. Potential workarounds:

- Increase the URI limit on the server (not currently possible, see https://github.com/yuzutech/kroki/issues/555)
- Use `kroki-fetch-diagram` to download (and save on the disk) images from the Kroki server. This extension will use a POST request to download images if the URI length is greater than `kroki-max-uri-length` (it's also possible to force POST requests using `kroki-http-method` attribute with the value `post`).

**Comment by lind** [comment]
Agree it would be fine to have a log message for this @Mogztter 

I have tried to find a way of intercepting the 414 response in Vertx but unfortunately haven't found a way so far. 
The `TooLongFrameException` from Netty is handled in [Http1xServerConnection](https://github.com/eclipse-vertx/vert.x/blob/master/src/main/java/io/vertx/core/http/impl/Http1xServerConnection.java#L505) but I don't find a way to add a interceptor or handler to the Vertx router in any way. Any idea?

**Comment by Mogztter** [comment]
> I have tried to find a way of intercepting the 414 response in Vertx but unfortunately haven't found a way so far.
The TooLongFrameException from Netty is handled in Http1xServerConnection but I don't find a way to add a interceptor or handler to the Vertx router in any way. Any idea?

In my opinion, we don't need to add an interceptor/handler server-side, a 414 status code is already explicit.
My idea was to log an explicit message in the Asciidoctor Kroki extension when the status code is 414.

**Comment by lind** [comment]
> In my opinion, we don't need to add an interceptor/handler server-side, a 414 status code is already explicit.
> My idea was to log an explicit message in the Asciidoctor Kroki extension when the status code is 414.

Of course! Jumped here from https://github.com/yuzutech/kroki/issues/555 so I was sill in Kroki world 😉

**Comment by ggrossetie** [comment]
Done in https://github.com/asciidoctor/asciidoctor-kroki/commit/0c55b63c29856b85b586b55a25137393be91bc17
